reframe/utility/sanity.py

```diff
@@ -530,28 +531,65 @@ def extractiter(patt, filename, tag=0, conv=None, encoding='utf-8'):
     the extracted values.
     '''
     for m in finditer(patt, filename, encoding):
-        try:
-            val = m.group(tag)
-        except (IndexError, KeyError):
-            raise SanityError(
-                "no such group in pattern `%s': %s" % (patt, tag))
-
-        try:
-            yield conv(val) if callable(conv) else val
-        except ValueError:
-            fn_name = '<unknown>'
+        if isinstance(tag, collections.Iterable) and not isinstance(tag, str):
+            val = []
+            for t in tag:
+                try:
+                    val.append(m.group(t))
+                except (IndexError, KeyError):
+                    raise SanityError(
+                        "no such group in pattern `%s': %s" % (patt, t))
+        else:
             try:
-                # Assume conv is standard function
-                fn_name = conv.__name__
-            except AttributeError:
+                val = m.group(tag)
+            except (IndexError, KeyError):
+                raise SanityError(
+                    "no such group in pattern `%s': %s" % (patt, tag))
+
+        if isinstance(val, list):
+            converted_vals = []
+            if not isinstance(conv, collections.Iterable):
+                conv = [conv] * len(val)
+
+            # Here we use the last conversion function for the remaining
+            # tags which don't have a corresponding one
+            for v, c in itertools.zip_longest(val, conv, fillvalue=conv[-1]):
                 try:
-                    # Assume conv is callable object
-                    fn_name = conv.__class__.__name__
+                    converted_vals.append(c(v) if callable(c) else v)
+                except ValueError:
+                    fn_name = '<unknown>'
+                    try:
+                        # Assume conv is standard function
+                        fn_name = c.__name__
+                    except AttributeError:
+                        try:
+                            # Assume conv is callable object
+                            fn_name = c.__class__.__name__
+                        except AttributeError:
+                            pass
```

Comment:
This should be abstracted in a separate function, e.g., `_callable_name()`, and will be used also in the current implementation.